pkg/tasks/monitoring_plugin.go

```diff
@@ -102,10 +103,19 @@ func (t *MonitoringPluginTask) Run(ctx context.Context) error {
 		if err = t.client.CreateOrUpdateService(ctx, svc); err != nil {
 			return fmt.Errorf("reconciling Console Plugin Service failed: %w", err)
 		}
-	}
 
-	{ // deployment
-		d, err := t.factory.MonitoringPluginDeployment()
+		secret, err := t.client.WaitForSecretByNsName(
+			ctx,
+			types.NamespacedName{
+				Namespace: svc.Namespace,
+				Name:      svc.Annotations["service.beta.openshift.io/serving-cert-secret-name"],
```

Comment:
@simonpasquier shouldn't we make sure CMO is listening for changes on that secret https://github.com/openshift/cluster-monitoring-operator/blob/a83f250fbf7efd2863a6faab471242322e563796/pkg/operator/operator.go#L665? otherwise the code that will change the sha may never run (if no other change triggers a sync).

Reply:
I missed the comment but yes you're probably right.